plugins/depthcamera/DepthCameraDriver.cpp

```diff
@@ -0,0 +1,273 @@
+{
+    return m_sensorData->m_height;
+}
+
+int DepthCameraDriver::getRgbWidth()
+{
+    return m_sensorData->m_width;
+}
+
+bool DepthCameraDriver::setRgbResolution(int width, int height)
+{
+    yCError(GZDEPTH) << "setRgbResolution: impossible to set resolution";
+    return false;
+}
+
+bool DepthCameraDriver::getRgbFOV(double& horizontalFov, double& verticalFov)
+{
+    if(!m_sensorData)
+    {
+        yCError(GZDEPTH) << "getRgbFOV: sensor data not available!";
+        return false;
+    }
+    horizontalFov = m_sensorData->horizontal_fov;
+    verticalFov   = m_sensorData->vertical_fov;
+    return true;
+}
+
+bool DepthCameraDriver::setRgbFOV(double horizontalFov, double verticalFov)
+{
+    yCError(GZDEPTH) << "setRgbFOV: impossible to set FOV";
+    return false;
+}
+
+bool DepthCameraDriver::getRgbMirroring(bool& mirror)
+{
+    mirror = false;
+    return true;
+}
+
+bool DepthCameraDriver::setRgbMirroring(bool mirror)
+{
+    yCError(GZDEPTH) << "setRgbMirroring: impossible to set mirroring";
+    return false;
+}
+
+bool DepthCameraDriver::getRgbIntrinsicParam(yarp::os::Property& intrinsic)
+{
+    if(!m_sensorData)
+    {
+        yCError(GZDEPTH) << "getRgbIntrinsicParam: sensor data not available!";
+        return false;
+    }
+    yarp::os::Value        rectM;
+
+    intrinsic.put("physFocalLength", 0.0);
+
+    intrinsic.put("focalLengthX",    m_sensorData->focalLengthX);
+    intrinsic.put("focalLengthY",    m_sensorData->focalLengthY);
+
+    intrinsic.put("k1",              m_sensorData->m_distModel.k1);
+    intrinsic.put("k2",              m_sensorData->m_distModel.k2);
+    intrinsic.put("k3",              m_sensorData->m_distModel.k3);
+    intrinsic.put("t1",              m_sensorData->m_distModel.p1);
+    intrinsic.put("t2",              m_sensorData->m_distModel.p2);
+    intrinsic.put("principalPointX", m_sensorData->m_distModel.cx);
+    intrinsic.put("principalPointY", m_sensorData->m_distModel.cy);
+    intrinsic.put("rectificationMatrix", rectM.makeList("1.0 0.0 0.0 0.0 1.0 0.0 0.0 0.0 1.0"));
+    intrinsic.put("distortionModel", "plumb_bob");
+    intrinsic.put("stamp",           m_sensorData->simTime);
+    return true;
+}
+
+bool DepthCameraDriver::getRgbImage(yarp::sig::FlexImage& rgbImage, yarp::os::Stamp* timeStamp)
+{
+    if(!m_sensorData)
+    {
+        yCError(GZDEPTH) << "getRgbImage: sensor data not available!";
+        return false;
+    }
+    if(!timeStamp)
+    {
+        yCError(GZDEPTH) << "getRgbImage: timestamp pointer invalid!";
+        return false;
+    }
+    std::lock_guard<std::mutex> lock(m_sensorData->m_mutex);
+    rgbImage.setPixelCode(m_sensorData->m_imageFormat);
+    rgbImage.resize(m_sensorData->m_width, m_sensorData->m_height);
+    memcpy(rgbImage.getRawImage(), m_sensorData->rgbCameraMsg.data().c_str(), m_sensorData->rgbCameraMsg.ByteSizeLong());
+    timeStamp->update(m_sensorData->simTime);
+    // TODO vertical and horizontal flip, display timestamp and time box
```

Comment:
Can you open an issue to track the TODO or remove it? TODO in the code are a bit difficult to track and manage.